### ISSUE TITLE
Stable xDS upstream names

### ISSUE THREAD

**Issue by kyessenov** [istio/istio#45503]
Follow-up to https://github.com/istio/istio/issues/45247.

EnvoyFilters and Wasm modules bypass the 1st class provider definition in Istio and address Envoy upstreams directly using cluster_name for side requests. Because the encoding of cluster names is not guaranteed to remain stable in Istio, this creates a risk that a Lua script or a Wasm binary might stop working during upgrades.

To solve this, we need to allow an explicit stable (alias) name for the created upstreams as part of the provider definition. This solves a couple of problems: 1) it's user defined and opt-in for the ServiceEntry and the generated provider upstreams, 2) we might be able to create multiple ones for the same hosts, 3) we can utilize xDS aliases to avoid breaking changes to the internal xDS names.

cc @howardjohn

**Comment by howardjohn** [comment]
Can we provide an abstraction that is not a cluster name? For example, in provider we have users reference a service hostname and we lookup a cluster based on that. Not sure how that translates into lua/wasm though.//

But I would prefer to not be tied to keeping xds internals stable as I don't think its an API we can commit to.

**Comment by kyessenov** [comment]
Wasm call uses cluster name: https://github.com/envoyproxy/envoy/blob/main/source/extensions/common/wasm/context.cc#L911

Lua uses cluster name:
https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/lua_filter#httpcall

ext* filters use cluster name for everything but Google gRPC (which doesn't support mTLS):
https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/core/v3/grpc_service.proto#config-core-v3-grpcservice-envoygrpc

Basically, the reality is that internal cluster names are used for all extension points, and we're leaking Istio names extensively. Envoy should support cluster aliases, if not, I can fix that.

**Comment by howardjohn** [comment]
I have no problem with cluster aliases. I would propose `outbound|port||host.com` becomes `host.com:port`?

Not sure if we want to support subsets though

**Comment by kyessenov** [comment]
I'm tempted to let the user name them - so that might mean Istio API extension to allow duplicate ServiceEntries for the same host. Either way we need to guide users.

**Comment by hzxuzhonghu** [comment]
Firstly `outbound|port||host.com` or `outbound|port|subset|host.com` are never changed since very beginning.

Second, if we allow setting arbitrary names additionally, will we have two clusters only with different names? Then some client side policy can not work well, since they are per cluster, right?
IMO, we cannot change the current cluster name, because envoyfilter is widely used.

**Comment by howardjohn** [comment]
It is not changing the name but adding an alias to allow referencing from
filters with a stable format. The cluster naming format actually has
changed a few times, in ambient and in ~1.10 - those changes happened to
only impact inbound though

**Comment by kyessenov** [comment]
https://github.com/envoyproxy/envoy/issues/28411
